### ISSUE TITLE
Virtual attributes?

### ISSUE THREAD

**Issue by ashaffer** [balderdashy/waterline#201]
Is waterline capable of handling virtual attributes?

E.g.
````javascript
{
fullName: function() {
    return this.firstName + ' ' + this.lastName;
  }
}
````

That can be accessed without calling a function, and delivered to the client as the value rather than an instance method.

**Comment by dmarcelino** [comment]
I would like to resurrect this feature request. Instance methods are nice but we lose them once we JSONfy the record. Ideally it would be great to have virtual attributes that could be exported to JSON.

@stevesims's suggestion in https://github.com/balderdashy/waterline/issues/201#issuecomment-30924443 looks good to me:
```
fullName: {
  type: 'virtual',
  set: function(val) {
    var name = val.split(' ');

    this.firstName = name[0];
    this.lastName = name[1];
  },
  get: function() {
    return this.firstName + ' ' + this.lastName;
  }
}
```

**Comment by devinivy** [comment]
+1 and I think the suggestion here looks great.

**Comment by tjwebb** [comment]
Here's one way to do this currently: https://github.com/tjwebb/sails-auth/blob/master/api/models/User.js#L24-L36

**Comment by dmarcelino** [comment]
@tjwebb, that's more or less the effect I would expect from a virtual attribute, I would just prefer a nice syntax! :)

**Comment by dmarcelino** [comment]
Given the discussion in #990, a possibly desirable feature of a virtual attribute is for it to behave as `transient` in Java. This would mean that if a user tried to save a record with a virtual attribute, waterline would strip it before persisting it in the datastore. This would ensure that a virtual attribute would remain virtual.

**Comment by sailsbot** [comment]
Thanks for posting, @ashaffer.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
